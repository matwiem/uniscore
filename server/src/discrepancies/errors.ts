export class InvalidFilterError extends Error {
    constructor (parameter: string, receivedValue: string, expected: string) {
        super(`Invalid Filter Error: invalid value for parameter '${parameter} - received ${receivedValue}, expected ${expected}'`)
    }
}
