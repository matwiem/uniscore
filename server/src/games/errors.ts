export class InvalidDTOFormat extends Error {
    constructor (message: string) {
        super(`Invalid DTO Format: ${message}`)
    }
}
