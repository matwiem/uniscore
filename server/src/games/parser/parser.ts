import { Game } from '@src/games/repository/repository'

export interface Parser {
    parse: (data: unknown) => Game
}
