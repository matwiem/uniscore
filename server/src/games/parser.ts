import { Game } from '@src/games/repository'

export interface Parser {
    parse: (data: unknown) => Game
}
