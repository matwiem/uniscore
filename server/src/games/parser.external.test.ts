import { InvalidDTOFormat } from './errors'
import { ParserExternal } from './parser.external'
import { gameDtoExternal } from './parser.external.mock'

describe('ParserExternal', () => {
    it('should parse game DTO', function () {
        const parser = new ParserExternal()
        const game = parser.parse(gameDtoExternal)
        expect(game).toMatchSnapshot()
    })

    it('should throw UnsupportedDTOFormat', function () {
        const parser = new ParserExternal()
        expect(() => {
            parser.parse({})
        }).toThrow(InvalidDTOFormat)
    })
})
