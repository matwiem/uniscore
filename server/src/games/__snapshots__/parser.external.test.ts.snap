// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParserExternal should parse game DTO 1`] = `
Game {
  "away": Team {
    "id": "ba27615c-07df-41f0-864b-332575f744f2",
    "players": {
      "62f377a6-ff87-4784-9631-05c555c4b14a": Player {
        "id": "62f377a6-ff87-4784-9631-05c555c4b14a",
        "statistics": {
          "rec": 2,
          "recYards": 39,
          "rushAttempts": 12,
          "rushTds": 0,
          "rushYdsGained": 73,
        },
      },
      "e11a2815-8f31-44b0-8c37-64240bcbc059": Player {
        "id": "e11a2815-8f31-44b0-8c37-64240bcbc059",
        "statistics": {
          "rec": null,
          "recYards": null,
          "rushAttempts": 14,
          "rushTds": 2,
          "rushYdsGained": 73,
        },
      },
      "f94fb8a3-cd77-4064-bc65-447550b43622": Player {
        "id": "f94fb8a3-cd77-4064-bc65-447550b43622",
        "statistics": {
          "rec": 9,
          "recYards": 47,
          "rushAttempts": null,
          "rushTds": null,
          "rushYdsGained": null,
        },
      },
    },
    "statistics": {
      "rec": 27,
      "recYards": 230,
      "rushAttempts": 37,
      "rushTds": 1,
      "rushYdsGained": 105,
    },
  },
  "home": Team {
    "id": "344e408b-846e-44ac-b491-7802ab0f7af0",
    "players": {
      "713daff7-86cc-4283-91a8-1579c3f58f59": Player {
        "id": "713daff7-86cc-4283-91a8-1579c3f58f59",
        "statistics": {
          "rec": 8,
          "recYards": 92,
          "rushAttempts": null,
          "rushTds": null,
          "rushYdsGained": null,
        },
      },
      "c90711f2-5c4e-4f2e-8635-0d3f2336f581": Player {
        "id": "c90711f2-5c4e-4f2e-8635-0d3f2336f581",
        "statistics": {
          "rec": null,
          "recYards": null,
          "rushAttempts": 2,
          "rushTds": 0,
          "rushYdsGained": -15,
        },
      },
      "e5b67a2e-e0f7-4646-a343-c9bd0ae447f0": Player {
        "id": "e5b67a2e-e0f7-4646-a343-c9bd0ae447f0",
        "statistics": {
          "rec": 1,
          "recYards": 4,
          "rushAttempts": 10,
          "rushTds": 0,
          "rushYdsGained": -4,
        },
      },
    },
    "statistics": {
      "rec": 31,
      "recYards": 257,
      "rushAttempts": 39,
      "rushTds": 0,
      "rushYdsGained": 90,
    },
  },
  "id": "15adf794-5630-4dc4-b0e7-f8d437b585b1",
  "statistics": {
    "attendance": 33876,
  },
}
`;
